### PR TITLE
feat(core): concrete dependency impl when exposing sociable dependencies with injection tokens

### PR DESCRIPTION
**Pull request by craig0990**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The pull request title follows the semantic commits guidelines: https://github.com/suites-dev/suites/blob/master/CONTRIBUTING.md#commit-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

1. `.sociable(ClassUnderTest).expose()`  does not accept string/symbol based dependency tokens
2. `.sociable(ClassUnderTest).expose(` does not expose concrete classes even if the parameter type is overidden as `never`

Issue Number: #464

## What is the new behavior?

* The `expose` method now accepts any `InjectableIdentifier` as the `dependency` and an optional `concreteImplementation: Type`
* `classesToExpose` is now a Map of `<Type | string | symbol, Type>` and is used to map the dependency identifier to a concrete value
* If the `dependency` is a class function (`Type`), use the dependency as both the identifier and the concrete value
    * Keeps backwards compatibility with previous behaviour
* Else, the `concreteImplementation` will be stored as the concrete value in the Map
* I've used method overloads to try and provide Intellisense/LSP hints that a `concreteImplementation` is required if the `dependency` is a symbol or string token
* This is all it takes for my [sample repository](https://github.com/craig0990/suites-sociable-inversify) from #464 to start passing

The rest of the changes are just to adapt to  `classesToExpose` now being a Map, and removing checks for exposed classes matching `typeof` "function"

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

Completely understand if this doesn't get merged for any reason; simply wanted to offer a possible solution to the issue raised.

### CODE REVIEW

**Comment by craig0990** [comment]
(Tweaked to fix the linting issues reported. Not sure why CI is failing. Tests pass, but test reports appear to have an issue 🤷🏻‍♂️ )

**Comment by omermorad** [comment]
@craig0990 there's an issue with the CI, let me check

File: packages/unit/src/types.ts [hunk 1]
```diff
@@ -29,6 +30,7 @@ export interface SociableTestBedBuilder<TClass> extends SociableTestBedBuilderCo
    * @param dependency
    */
   expose(dependency: Type): SociableTestBedBuilder<TClass>;
+  expose(dependency: symbol | string, concreteImplementation: Type): SociableTestBedBuilder<TClass>;
```
Comment (omermorad) [inline packages/unit/src/types.ts:33]: `expose(dependency: symbol | string, implementation: Type): SociableTestBedBuilder<TClass>`?

Comment (omermorad) [reply packages/unit/src/types.ts:33]: Commented here: https://github.com/suites-dev/suites/issues/464#issuecomment-2438499353

**Pull request closed without merging**